src/ui/components/SecondaryToolbox/ReduxDevTools.tsx

```diff
@@ -59,6 +66,94 @@ export const ReduxDevToolsPanel = () => {
   );
 };
 
+interface PointWithLocation {
+  location: Location;
+  point?: TimeStampedPoint;
+}
+
+const reduxDispatchJumpLocationCache = createCache<
+  [
+    replayClient: ReplayClientInterface,
+    point: ExecutionPoint,
+    time: number,
+    sourcesState: SourcesState
+  ],
+  PointDescription | undefined
+>({
+  config: { immutable: true },
+  debugLabel: "ReduxDispatchJumpLocation",
+  getKey: ([replayClient, point, time, sourcesState]) => point,
+  load: async ([replayClient, point, time, sourcesState]) => {
+    const pauseId = await pauseIdCache.readAsync(replayClient, point, time);
+    const frames = (await getPauseFramesAsync(replayClient, pauseId, sourcesState)) ?? [];
+    const filteredPauseFrames = frames.filter(frame => {
+      const { source } = frame;
+      if (!source) {
+        return false;
+      }
+
+      // Filter out everything in `node_modules`, so we have just app code left
+      // TODO There may be times when we care about renders queued by lib code
+      // TODO See about just filtering out React instead?
+      return !MORE_IGNORABLE_PARTIAL_URLS.some(partialUrl => source.url?.includes(partialUrl));
+    });
+
+    // The first 2 elements in filtered pause frames are from replay's redux stub, so they should be ignored
+    // The 3rd element is the user function that calls it, and will most likely be the `dispatch` call
+    const preferredFrame = filteredPauseFrames[2];
```

Comment:
:hand: I think we need to update the logic here as well.

The issue is that there could be multiple Redux middleware that are in the call stack, on top of the actual `store.dispatch()` call.

As an example, clicking "Jump" for one of the Redux actions in our own FE E2E tests is jumping me to our "context middleware":

![image](https://github.com/replayio/devtools/assets/1128784/c2b5a763-3b0a-49df-83cd-0cdbc325c43f)

If we look at the call stack for this case, where we _really_ want to end up is the actual place where `dispatch(someAction())` was called - in this case, one of our store setup files:

![image](https://github.com/replayio/devtools/assets/1128784/54102ec2-98f5-404f-8349-536d3d35272b)

Similarly, if I jump to the line for a `locationSelected` action, it would be best if we could end up jumping to the `dispatch(locationSelected())` line in this thunk:

![image](https://github.com/replayio/devtools/assets/1128784/71d4bc58-6a34-43dc-afb4-cafea71a2c25)


I don't yet know a good heuristic for figuring out how many frames we need to skip backwards.  We know that a couple frames are part of Redux itself, but there could be multiple middleware frames before that on the call stack too.

Can you do some experimenting with our app and a couple other Redux apps and try to work out a good heuristic?

Reply:
Nice find. My hunch is that it'd be nice to land this and then have a more dedicated issue/pr for investigating middleware. 

For instance, I could see this tying into an ignorelist feature for files if we can't come up with the best heuristic.

Also - landing this change will make it easier to see what other cases we want to improve as well

Reply:
I agree, tracing the dispatch from middleware will require more work, so it is probably better to make it into a dedicated pr and land this as is. I will try experimenting with babel to find a fix

Reply:
Yeah, landing in the middleware is a step in the right direction, and admittedly not all Redux apps are going to have a userland middleware.